recipes/resiprocate/all/conanfile.py

```diff
@@ -1,94 +1,104 @@
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=1.53.0"
 
-required_conan_version = ">=1.29.1"
 
 class ResiprocateConan(ConanFile):
     name = "resiprocate"
-    description = "The project is dedicated to maintaining a complete, correct, and commercially usable implementation of SIP and a few related protocols. "
-    topics = ("sip", "voip", "communication", "signaling")
-    url = "https://github.com/conan-io/conan-center-index"
-    homepage = "http://www.resiprocate.org"
+    description = (
+        "The project is dedicated to maintaining a complete, correct, "
+        "and commercially usable implementation of SIP and a few related protocols."
+    )
     license = "VSL-1.0"
-    settings = "os", "compiler", "build_type", "arch"
-    options = {"fPIC": [True, False],
-               "shared": [True, False],
-               "with_ssl": [True, False],
-               "with_postgresql": [True, False],
-               "with_mysql": [True, False]}
-    default_options = {"fPIC": True,
-                       "shared": False,
-                       "with_ssl": True,
-                       "with_postgresql": True,
-                       "with_mysql": True}
-    _autotools = None
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/resiprocate/resiprocate/wiki/"
+    topics = ("sip", "voip", "communication", "signaling")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_ssl": [True, False],
+        "with_postgresql": [True, False],
+        "with_mysql": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ssl": True,
+        "with_postgresql": True,
+        "with_mysql": True,
+    }
 
     def config_options(self):
-        if self.settings.os == 'Windows':
+        if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
-        if self.settings.os in ("Windows", "Macos"):
+        if self.settings.os == "Windows" or is_apple_os(self):
             # FIXME: Visual Studio project & Mac support seems available in resiprocate
-            raise ConanInvalidConfiguration("reSIProcate recipe does not currently support {}.".format(self.settings.os))
+            raise ConanInvalidConfiguration(
+                f"reSIProcate recipe does not currently support {self.settings.os}."
+            )
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.with_ssl:
-            self.requires("openssl/1.1.1q")
+            self.requires("openssl/[>=1.1 <4]")
```

Comment:
Didn't you say in a previous commit that it was not compatible with openssl 3.x?

Reply:
But because pre-built packages of libmysqlclient are built against openssl 3.x, you'll have to disable with_ssl or with_mysql option in this recipe, otherwise it will never pass c3i due to missing packages.